src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -176,4 +192,18 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
             }
         }
     }
+
+    private static bool SchemaRepresentsObject(JsonElement schemaElement)
+    {
+        foreach (var property in schemaElement.EnumerateObject())
```

Comment:
This will throw if `schemaElement.ValueKind` is not `Object`.
```suggestion
        if (schemaElement.ValueKind is not JsonValueKind.Object)
        {
             return false;
        }

        foreach (var property in schemaElement.EnumerateObject())
```